cpp/dolfinx/la/SparsityPattern.cpp

```diff
@@ -224,11 +198,48 @@ void SparsityPattern::insert_diagonal(const std::vector<int32_t>& rows)
   }
 }
 //-----------------------------------------------------------------------------
+std::shared_ptr<const common::IndexMap>
+SparsityPattern::index_map(int dim) const
+{
+  return _index_maps.at(dim);
+}
+//-----------------------------------------------------------------------------
+std::vector<std::int64_t> SparsityPattern::column_indices() const
+{
+  if (!_graph)
+    throw std::runtime_error("Sparsity pattern has not been finalised.");
+
+  std::array range = _index_maps[1]->local_range();
+  const std::int32_t local_size = range[1] - range[0];
+  const std::int32_t num_ghosts = _col_ghosts.size();
+  std::vector<std::int64_t> global(local_size + num_ghosts);
+  std::iota(global.begin(), std::next(global.begin(), local_size), range[0]);
+  std::copy(_col_ghosts.begin(), _col_ghosts.end(),
+            global.begin() + local_size);
+  return global;
+}
+//-----------------------------------------------------------------------------
+common::IndexMap SparsityPattern::column_index_map() const
+{
+  if (!_graph)
+    throw std::runtime_error("Sparsity pattern has not been finalised.");
+
+  std::array range = _index_maps[1]->local_range();
+  const std::int32_t local_size = range[1] - range[0];
+  return common::IndexMap(
+      _comm.comm(), local_size,
+      dolfinx::MPI::compute_graph_edges(
```

Comment:
@chrisrichardson `compute_graph_edges` should be avoided whenever possible. It involves global all-to-all communication. We must have computed the required information already during construction?

Reply:
I am very happy to look at this in a future PR (soon).